### DISCUSSION TITLE
Dynamic Visual Studio path detection

### DISCUSSION THREAD

**Discussion by silent-tech** [Pagghiu/SaneCppLibraries#16] [Ideas]
Hello Pagghiu,

I found a minor issue: one of my test machines have a VS2022 that is not installed into the default directory. So, Tools and Plugin module couldn't find it. I made an alternate version that uses auto detection instead of constant strings. You can see my solution in these files:
https://github.com/silent-tech/SaneCppLibraries/blob/main/Tools/Tools.bat
https://github.com/silent-tech/SaneCppLibraries/blob/main/Libraries/Plugin/Plugin.cpp

Currently, it finds the latest version, but it can be modified to find specific versions (2019, 2022) as well.

Please review it when you have some free time. If you like it, I would send a pull request to integrate it to the main version.

**Comment by Pagghiu** [comment]
Thanks @silent-tech!

Finding the latest visual studio version is fine, and it's exactly what it's being hardcoded now anyway.
I would place the `VisualStudioPathFinder` class as is into a `Plugin/Internal/VisualStudioPathFinder.h` file.

Feel free to post the MR, I can review it more in detail and that would be very helpful as it would be the first real PR of the project.
I am not even sure if CI is setup correctly for Pull Requests (I've never tried it), so this could be helpful to make sure that it actually works as expected 😅

**Comment by silent-tech** [comment]
Unfortunately, I violated some minor clang formatting rules, and I didn't check them before I committed my changes. Sorry, I fixed them in [d52556b](https://github.com/Pagghiu/SaneCppLibraries/pull/18/commits/d52556bdced7f4f2bb1676285f1081e4ca57f26d).

**Reply by Pagghiu** [reply]
Yes, you can just run `./SC.sh format` and that will take care of it for you (as described [here](https://pagghiu.github.io/SaneCppLibraries/page_coding_style.html#autotoc_md236)), but I assume you've already been discovering it 😌 .
I will start reviewing your PR right now.
